libcst/metadata/scope_provider.py

```diff
@@ -616,7 +616,10 @@ def _gen_dotted_names(
             # this is not an import
             return
         name_values = iter(_gen_dotted_names(value))
-        (next_name, next_node) = next(name_values)
+        next_pair = next(name_values, None)
+        if next_pair is None:
+            return
+        (next_name, next_node) = next_pair
```

Comment:
We just need try catch on this line. Then we can save the `next` in `next_pair = next(name_values, None)`.